assignments/prototypes.js

```diff
@@ -32,16 +52,48 @@
   * should inherit destroy() from GameObject through CharacterStats
   * should inherit takeDamage() from CharacterStats
 */
+
+function Humanoid({ createdAt, dimensions, healthPoints, name, team, weapons, language }) {
+  CharacterStats.call(this, createdAt, name, dimensions, healthPoints);
+  this.team = team;
+  this.weapons = weapons;
+  this.language = language;
+
+}
+Humanoid.prototype = Object.create(CharacterStats.prototype);
+Humanoid.prototype.greet = function () {
+  return `${this.name} offers a greeting in ${this.language}`
+
+}
  
 /*
   * Inheritance chain: GameObject -> CharacterStats -> Humanoid
   * Instances of Humanoid should have all of the same properties as CharacterStats and GameObject.
   * Instances of CharacterStats should have all of the same properties as GameObject.
 */
 
+// stretch new objects
+function Villain(villianDetails) {
+  Humanoid.call(this, villianDetails);
+
+  Villain.prototype.move = function (characterName) {
+    return (characterName.healthPoints - 5);
+  }
+
+};
+
+function Hero(heroDetails) {
+  Villain.call(this, heroDetails);
+
+  Hero.prototype.ultimate = function (characterName) {
+    return characterName.healthPoints - 30;
+  }
+
+};
```

Comment:
These prototype methods should probably be defined outside of the function's inner scope here. Otherwise, you'd be reassigning the prototype method on every instantiation.